src/vtab/mod.rs

```diff
@@ -57,6 +57,19 @@ use crate::{str_to_cstring, Connection, Error, InnerConnection, Result};
 // ffi::sqlite3_vtab => VTab
 // ffi::sqlite3_vtab_cursor => VTabCursor
 
+/// Virtual table kind
```

Comment:
I think this needs better docs. Can you use this? Do a quick read-through of it to make sure I didn't make any obvious mistakes, also.

```rs
/// Virtual table kind, indicating whether or not the table is
/// [eponymous](https://sqlite.org/vtab.html#eponymous_virtual_tables).
#[non_exhaustive]
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum VTabKind {
    /// Non-eponymous tables, which are the most flexible.
    Default,
    /// Eponymous tables exist implicitly in the main schema of the database,
    /// and do not require setup such as `CREATE VIRTUAL TABLE` to be used.
    ///
    /// These should implement [`CreateVTab::create`] and [`VTab::connect`]
    /// identically. (Which is the default behavior if [`CreateVTab::create`] is
    /// not explicitly overridden)
    Eponymous,
    /// Eponymous-only tables are tables which may only be used eponymously, and
    /// cannot be used with `CREATE VIRTUAL TABLE`.
    ///
    /// This is often useful for table valued functions, although note that this
    /// prevents use as a temporary table, in patterns like the following:
    ///
    /// ```sql
    /// CREATE VIRTUAL TABLE temp.something USING my_vtab(args, here);
    /// ```
    ///
    /// But whether or not this pattern makes sense to support depends on the
    /// implementation of your table.
    ///
    /// Eponymous-only tables will not have the [`CreateVTab::create`] and
    /// [`CreateVTab::destroy`] methods called. [`VTab::connect`] will be used
    /// instead (and cleanup will be performed by [`Drop::drop`]).
    ///
    /// If the SQLite version is older than 3.9.0, `rusqlite` will produce an
    /// error when you attempt to register a `EponymousOnly` virtual table
    EponymousOnly,
}
```